hs_collection_resource/utils.py

```diff
@@ -128,13 +128,16 @@ def get_collectable_resources(user, coll_resource):
     get_my_resources_list(user)
 
     # resource is collectable if
-    # 1) Shareable=True
-    # 2) OR, current user is a owner of it
-    # 3) exclude this resource as well as resources already in the collection
+    # 1) shareable=True and resource is public in my resources, --or--
+    # 2) shareable=True and resource is accessible with view privilege, --or--
+    # 2) current user owns the resource.
+    # Also exclude this resource as well as resources already in the collection
+    # Start with both my resources and annexed public resources
     return get_my_resources_list(user) \
+        .filter(Q(raccess__shareable=True) |  # shareable and public or viewable, --or--
+                Q(r2urp__user=user, r2urp__privilege=PrivilegeCodes.OWNER)) \
         .exclude(short_id=coll_resource.short_id) \
-        .exclude(id__in=coll_resource.resources.values_list("id", flat=True)) \
-        .filter(Q(raccess__shareable=True) | (Q(r2urp__user=user) & Q(r2urp__privilege=PrivilegeCodes.OWNER)))
+        .exclude(id__in=coll_resource.resources.values_list("id", flat=True))
```

Comment:
Maybe it's just getting late, but I'm not seeing how this would be faster, but I'll trust this is faster.  Doesn't seem like it would be any slower

Reply:
It's not any faster, it just seems more stylistically consistent to do filters first and then excludes.